netdata/netdata
# eBPF OOM kill tracking

##### Summary

This PR brings eBPF-based OOM kill tracking. https://github.com/netdata/product/issues/2158

##### Component Name

ebpf.plugin

##### Test Plan

- Compile this branch.
- Turn off swap, e.g. `sudo swapoff -a`.
- Run netdata with `sudo choom -n -1000 -- sudo netdata -D` (so it doesn't get OOM killed).
- Run something that will exhaust memory quickly, e.g. `sudo choom -n 1000 -- tail /dev/zero`.
- Check that the chart shows up (`apps.oomkills`).

Tested with:

- Ubuntu 18.04.1 (Linux 4.15)
- Ubuntu 21.04 (Linux 5.11.0-31)
